### ISSUE TITLE
hide unnecessary steps

### ISSUE THREAD

**Issue by mPokornyETM** [jenkinsci/pipeline-graph-view-plugin#145]
### What feature do you want to see added?

There are many steps, which does not produce any outputs (or it is unusable)

Like in this example. I care only about the output from system command.

![image](https://user-images.githubusercontent.com/89339813/193458108-b4283196-a9ba-45a3-bff8-2d7f3d8b9b90.png)

```

stage('Node info') {
    node() {
      sleep 1
      Map opts = [label : 'print node info', script : 'hostname']
      if (isUnix()) {
          sh opts
      } else {
          bat opts
      }
      deleteDir();
    }
}
```

So it will be very useful, when I can filter out unnecessary steps. This shall be somewhere configurable like:
+ who cares (any, admin, or specific user settings)
+ hide / show steps with empty output
+ hide / show specific steps (like in my example sleep, it produce output, but in many cases no body [or only admin] cares)

this improves daily workflow extremely. When I can shows for 'normal' users only relevant output and for jenkins--administrators some more steps. Anyway when we need analyze some special issues we switch to console-log (and the fun job starts ;-) )

**Comment by timja** [comment]
I don't think we want to hide steps, but maybe we can style steps with no output differently so that you know there's no point clicking them?

**Comment by mPokornyETM** [comment]
In that case just add include / exclude filter there. And we can set it programmatically in our pipe code as well.
So win win situation, because filtering in case of multiple lines is in any case good idea.

**Comment by avlec** [comment]
The ability to control hiding steps that are skipped via `when` conditions might also be valuable.
